src/components/touchspin/touchspin.js

```diff
@@ -0,0 +1,191 @@
+    }
+
+    this.onUpdateValue({ target: { value: updatedValue } });
+  };
+
+  onUpdateValue = event => {
+    const { maxValue, minValue } = this.state;
+    const { name, onChange } = this.props;
+    const { value } = event.target;
+
+    let parsedValue = Number.parseInt(value, 10);
+    parsedValue = Number.isNaN(parsedValue) ? minValue : parsedValue;
+
+    const maxDisabled = parsedValue > maxValue;
+    const minDisabled = parsedValue < minValue;
+
+    clearTimeout(this.timer);
+
+    this.setState({
+      displayValue: value,
+      value: parsedValue,
+      maxDisabled,
+      minDisabled
+    });
+
+    const mockTarget = {
+      ...event.target,
+      name,
+      value: parsedValue
+    };
+
+    const mockEvent = {
+      ...mockTarget,
+      target: { ...mockTarget },
+      currentTarget: { ...mockTarget },
+      persist: helpers.noop
+    };
+
+    return onChange(mockEvent);
+  };
+
+  render() {
+    const { displayValue, maxDisabled, minDisabled } = this.state;
+    const { className, labelMax, labelMaxDescription, labelMin, labelMinDescription, name } = this.props;
+
+    return (
+      <InputGroup className={`cloudmeter-touchspin ${className}`}>
+        <InputGroup.Button>
+          <Button
+            onClick={this.onMin}
+            onMouseDown={this.onFocusMin}
+            onMouseUp={this.onBlur}
+            onMouseLeave={this.onBlur}
+            className="cloudmeter-touchspin-min-button form-control"
+            aria-hidden
+            tabIndex={-1}
+            title={labelMinDescription}
+            disabled={minDisabled}
+          >
+            {labelMin}
+          </Button>
+        </InputGroup.Button>
+        <Form.FormControl
+          type="text"
+          name={name}
+          value={displayValue}
+          className="cloudmeter-touchspin-input"
+          onChange={this.onUpdateValue}
+        />
+        <InputGroup.Button>
+          <Button
+            onClick={this.onMax}
+            onMouseDown={this.onFocusMax}
+            onMouseUp={this.onBlur}
+            onMouseLeave={this.onBlur}
+            className="cloudmeter-touchspin-max-button form-control"
+            aria-hidden
+            tabIndex={-1}
+            title={labelMaxDescription}
+            disabled={maxDisabled}
+          >
+            {labelMax}
+          </Button>
+        </InputGroup.Button>
+      </InputGroup>
+    );
+  }
+}
+
+TouchSpin.propTypes = {
+  className: PropTypes.string,
+  labelMax: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  labelMaxDescription: PropTypes.string,
+  labelMin: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  labelMinDescription: PropTypes.string,
+  maxValue: PropTypes.number,
+  minValue: PropTypes.number,
+  name: PropTypes.string.isRequired,
+  onChange: PropTypes.func,
+  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+};
+
+TouchSpin.defaultProps = {
+  className: '',
+  labelMax: <Icon type="fa" name="plus" />,
```

Comment:
might consider an sr-only span to help with accessibility

Reply:
updated: so it looks like we were already hiding the buttons for min and max and just exposing the input field with a `aria-hidden`. The field can still be wrapped with a label, so thinking we're safe, least for pf3 accessibility

What was broken for sure was the ability to be notified when the field validation updated inside the dialog. So went back and added the `aria-live` attributes to the two most recent parts we updated... the Add Source Wizard, and the Scans dialog covered by this PR

Reply:
... and yeahhhhhh its a little sad, on the upside removing the subsequent ajax calls the expanded state was forced to handle did improve overall load time